system-addon/content-src/components/TopSites/TopSiteForm.jsx

```diff
@@ -123,17 +123,27 @@ export class TopSiteForm extends React.PureComponent {
     return url;
   }
 
-  validateUrl(url) {
+  _tryParseUrl(url) {
     try {
-      return !!new URL(this.cleanUrl(url));
+      return new URL(url);
     } catch (e) {
       return false;
     }
   }
 
+  validateUrl(url) {
+    const validProtocols = ["http:", "https:"];
+    const urlObj = this._tryParseUrl(url) || this._tryParseUrl(`https://${url}`);
+
+    return urlObj && validProtocols.includes(urlObj.protocol);
+  }
+
   validateCustomScreenshotUrl() {
+    const invalidProtocols = ["file:"];
     const {customScreenshotUrl} = this.state;
-    return !customScreenshotUrl || this.validateUrl(customScreenshotUrl);
+
+    return !customScreenshotUrl ||
+      (this.validateUrl(customScreenshotUrl) && !invalidProtocols.some(p => customScreenshotUrl.startsWith(p)));
```

Comment:
`invalidProtocols` is redundant code now. `validateUrl` will only be truthy if it's http/s, so the `.some` check will always be `false`.